pkg/operator/ceph/object/controller.go

```diff
@@ -142,9 +144,67 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		}
 	}
 
+	// Watch Secrets secrets annotated for the object store
+	err = c.Watch(source.Kind[client.Object](mgr.GetCache(),
+		&corev1.Secret{TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: corev1.SchemeGroupVersion.String()}},
+		handler.EnqueueRequestsFromMapFunc(mapSecretToCR),
+		secretPredicate()))
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// Watch update and create secrets annotated with the object store name
+func secretPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			secret, ok := e.ObjectNew.(*corev1.Secret)
+			if !ok {
+				return false
+			}
+			annotations := secret.GetAnnotations()
+			return annotations[objectStoreDependentResourceAnnotation] != ""
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			secret, ok := e.Object.(*corev1.Secret)
+			if !ok {
+				return false
+			}
+			annotations := secret.GetAnnotations()
+			return annotations[objectStoreDependentResourceAnnotation] != ""
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false // Ignore deletes for simplicity
+		},
+	}
+}
+
+// Maps secret annotated with the object store name to the object store CR
+func mapSecretToCR(ctx context.Context, obj client.Object) []reconcile.Request {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return nil
+	}
+
+	value, exists := secret.GetAnnotations()[objectStoreDependentResourceAnnotation]
+	if !exists || value == "" {
+		return nil
+	}
+	objStoreNames := strings.Split(value, ",")
+	requests := make([]reconcile.Request, len(objStoreNames))
+	for i, objStoreName := range objStoreNames {
+		requests[i] = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      objStoreName,
```

Comment:
Just in case a user edits this, or some logic changes in the future, trimming space will make sure we handle lists where items have comma-space-separated elements like `store1, store2,store3`

```suggestion
				Name:      strings.TrimSpace(objStoreName),
```

Reply:
done